### DISCUSSION TITLE
OrthographicView on PyDeck

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5291] [Q&A]
Hi, I wanted to know if it is possible to use the OrthographicView in PyDeck. I want to visualize some non geospatial data.

**Comment by ajduberstein** [comment]
All the views are there, e.g. [FirstPersonView](https://github.com/visgl/deck.gl/issues/4954). Here's an OrthographicView example:

```python
import pydeck
import math

# Plot of a square root function

plot_data = [
    {
        'x': num,
        'y': math.sqrt(num),
        'pos': [num, -math.sqrt(num)] # Note that the y position is negative
    } for num in range(0, 10000)
]

point_cloud_layer = pydeck.Layer(
    "ScatterplotLayer",
    data=plot_data,
    get_position="pos",
    radius_unit='"pixels"',  # Many deck.gl layers default to radius units in meters
    auto_highlight=True,
    get_fill_color=[255, 0, 0],
    highlight_color=[255, 140, 255],
    pickable=True,
    get_radius=0.5,
    radius_min_pixels=0.5
)

# Create an x and y axis
axis_layer = pydeck.Layer(
    "PathLayer",
    data=[
        {'path': [[-100000, 0], [100000, 0]]},
        {'path': [[0, -100000], [0, 100000]]}
    ],
    get_path="path",
    width_unit='"pixels"',
    get_color=[0, 0, 0],
    width_min_pixels=1,
    width_max_pixels=1,
    get_width=1,
)

# Add labels to x and y axis
label_data = [{"x": i, "y": 0, "label": str(i)} for i in range(-1000, 1000) if i % 10 == 0]
label_data += [{"x": 0, "y": i, "label": str(-1 * i)} for i in range(-1000, 1000) if i % 10 == 0]

axis_labels = pydeck.Layer(
    "TextLayer",
    data=label_data,
    get_position=['x', 'y'],
    get_size=15,
    size_max_pixels=15,
    size_min_pixels=15,
    get_text='label',
    get_alignment_baseline='"bottom"',
)

view_state = pydeck.ViewState(target=[0, 0, 0], zoom=2)
view = pydeck.View(
    type="OrthographicView",
    controller=True
)

r = pydeck.Deck(
    [point_cloud_layer, axis_layer, axis_labels],
    map_provider=None,
    initial_view_state=view_state,
    views=[view]
)
r.to_html(css_background_color='skyblue')
```

![image](https://user-images.githubusercontent.com/2204757/103464758-0d5a9d80-4ceb-11eb-925e-794243a84f6e.png)

**Reply by cornhundred** [reply]
Thanks @ajduberstein!! I'll let you know if I have any issues

**Comment by cornhundred** [comment]
Hi @ajduberstein, is there a way to get different layers to have different zoom behaviors? For instance, I would want to have the x and y axes each zoom in the horizontal and vertical directions, respectively. 

Also, is there a way to reverse the y orientation convention (e.g. positive y values go downwards)?

Also, I ran into a few bugs where nothing is plotted if the names of data points have spaces or dashes in them - I'll raise this as an issue later and try to see if I can figure out what is causing this in the code.

**Reply by Pessimistress** [reply]
An option to flip the y axis will be added in the next release: https://github.com/visgl/deck.gl/pull/5145

**Reply by ajduberstein** [reply]
> different layers to have different zoom behaviors

I don't believe there's a functioning `onViewStateChange` within pydeck if you're outside of Jupyter. If you're in Jupyter, you would have the option to use [one of the event handlers](https://pydeck.gl/event_handling.html#view-state-change-events-with-on-view-state-change). A custom layer here might also be easy to implement--you wouldn't have the issue with an external dependency like we did with nebula.

> spaces and dashes

I'm surprised this hasn't come up sooner--it's a bug. I've added an issue (#5308).

**Reply by cornhundred** [reply]
Thanks @ajduberstein, I'll give that a shot.

I'm remembering now that having NaNs in the data also cause a problem - I can raise issue later.
